app/src/main/java/com/hedvig/app/service/badge/NotificationBadge.kt

```diff
@@ -0,0 +1,28 @@
+package com.hedvig.app.service.badge
+
+import androidx.datastore.preferences.core.Preferences
+import androidx.datastore.preferences.core.booleanPreferencesKey
+import com.hedvig.android.owldroid.type.TypeOfContract
+
+sealed class NotificationBadge(val key: Preferences.Key<Boolean>) {
+    object CrossSellInsuranceFragmentCard : NotificationBadge(
+        booleanPreferencesKey("SEEN_CROSS_SELL_INSURANCE_FRAGMENT_CARD")
+    )
+
+    sealed class BottomNav(key: Preferences.Key<Boolean>) : NotificationBadge(key) {
+        object CrossSellOnInsuranceFragment : BottomNav(booleanPreferencesKey("SEEN_CROSS_SELLS"))
+    }
+
+    companion object {
+        fun fromPotentialCrossSells(
+            potentialCrossSells: Set<TypeOfContract>
+        ): List<BottomNav> =
+            potentialCrossSells.mapNotNull { typeOfContract ->
+                if (typeOfContract == TypeOfContract.SE_ACCIDENT) {
```

Comment:
This breaks the contract that we set out for the cross-sell tab notification: this only checks for `SE_ACCIDENT`, but the business rule is that we should show the tab notification any time there is a cross-sell available that the member has not seen 🙂

Reply:
Ugh okay, I misunderstood this a bit, let me go back and think about it a bit again. Might have dug a bit of a hole for myself.